content/post/2024-02-29-stepping-down.md

```diff
@@ -0,0 +1,58 @@
+---
+date: 2024-02-29
+title: Stepping down as a (rust-) libp2p maintainer
+tags: [tech, libp2p]
+---
+
+I don't see myself making major contributions to (rust-) libp2p in the near future and thus I am stepping down as a maintainer.
+
+As announced before, [I have left Protocol Labs in December 2023](https://github.com/libp2p/rust-libp2p/discussions/5007).
+After a 2 month re-orientation break, I have decided to move on entirely.
+
+My first commit was 5y ago, [a small bug fix in our address handling](https://github.com/libp2p/rust-libp2p/pull/1204).
+Since then lots happened. A couple of milestones I was involved in:
+
+- First major contribution was an [implementation of S-Kademlia's disjoint paths](https://discuss.libp2p.io/t/s-kademlia-lookups-over-disjoint-paths-in-rust-libp2p/571).
+  While a good first project, it was never deployed on any of the larger networks.
+  Attacking a DHT is [not that hard](https://arxiv.org/pdf/2307.12212.pdf), making it secure takes a lot more than this.
+- Eventually we added what one would expect every p2p library to have - support for hole punching.
+  This was a huge effort.
+  We rolled our own relay implementation, a custom coordination protocol, various tricks around address discovery and mapping, ...
+  Long story short, [we invented lots of new wheels](https://blog.ipfs.tech/2022-01-20-libp2p-hole-punching/) ...
+  Really, we should have started off of [ICE](https://en.wikipedia.org/wiki/Interactive_Connectivity_Establishment).
```

Comment:
```suggestion
  In hindsight, it might have been better to ...
  Really, we should have started off of [ICE](https://en.wikipedia.org/wiki/Interactive_Connectivity_Establishment).
```

You are saying "started off" which implies you wanted to modify it to fit your needs? Perhaps add a few more thoughts here on why you think that is and how would have liked to make it work.